src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -959,6 +960,30 @@ public virtual TokenValidationResult ValidateToken(string token, TokenValidation
             if (tokenParts.Length != JwtConstants.JwsSegmentCount && tokenParts.Length != JwtConstants.JweSegmentCount)
                 return new TokenValidationResult { Exception = LogHelper.LogExceptionMessage(new ArgumentException(LogHelper.FormatInvariant(LogMessages.IDX14111, token))), IsValid = false };
 
+            if (validationParameters.ConfigurationManager != null)
+            {
+                try
+                {
+                    validationParameters.Configuration = validationParameters.ConfigurationManager.GetBaseConfigurationAsync(CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
This method performs validation but it also updates the validationParameters.Configuration, is that the desired behavior? Typically validation methods do not change data.

Reply:
Ahhh, that's a valid concern. I've changed the logic so that we clone the TVP before we set the configuration property on it. Does that address this particular concern?